src/components/Charts/Charts.tsx

```diff
@@ -1,4 +1,95 @@
-export default function Charts() {
-  // RENDER
-  return <div className="Charts">Charts</div>;
+// Libs
+import * as React from "react";
+import { Container } from "@material-ui/core";
+import {
+	ResponsiveContainer,
+	CartesianGrid,
+	XAxis,
+	YAxis,
+	LineChart,
+	Line,
+	Tooltip,
+	Legend,
+} from "recharts";
+
+// Assets
+import { configMonths, configGroups, iTransaction } from "../../types";
+
+export default function Charts({
+	cleanTransactions,
+}: {
+	cleanTransactions: Array<iTransaction>;
+}) {
+	// Definitions
+	const [chartData, setChartData] = React.useState<Array<{}>>([{}]);
+
+	React.useEffect(() => {
+		const groupsObj = configGroups.reduce(
+			(passed: { [group: string]: number }, group) => {
+				passed[group.name] = 0;
+				return passed;
+			},
+			{}
+		);
+		const newChartData: Array<{
+			month: string;
+			[group: string]: number | string;
+		}> = configMonths.map((monthKey) => ({
+			month: monthKey,
+			...groupsObj,
+		}));
+
+		const catGroupMap: { [category: string]: string } = {};
+		for (const group of configGroups) {
+			for (const category of group.categories) {
+				catGroupMap[category] = group.name;
+			}
+		}
+
+		for (const transaction of cleanTransactions) {
+			const monthIndex = parseInt(transaction.date.split(".")[1]) - 1;
+			const currentVal = newChartData[monthIndex][
+				catGroupMap[transaction.category]
+			] as number;
+			newChartData[monthIndex][catGroupMap[transaction.category]] =
+				currentVal + transaction.amount;
+		}
+
+		setChartData(newChartData);
```

Comment:
une raison pour laquelle tu as utilisé un `useEffect` et un `useState` (au lieu d'un `useMemo`?)

Reply:
Hooks removed in future PR – charts data generated  upon each render, not stored.